src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -104,6 +104,8 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
 
+        } catch (FileNotFoundException fnfe) {
```

Comment:
this will probably never be encountered. 
the current implementation always creates the file if it is not found. 
you need to check for the existence of the file before this `try ... catch` block